### PR TITLE
Fix default acm to empty list if key is present but set to None

### PR DESCRIPTION
**Pull request by gonzalocasas**
### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.rst` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.Configuration`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py [hunk 1]
```diff
@@ -122,10 +122,11 @@ def plan_motion_async(self, callback, errback,
...
             joint_state, MultiDOFJointState(header=header), is_diff=True)
-        attached_collision_meshes = options.get('attached_collision_meshes', [])
-        for acm in attached_collision_meshes:
-            aco = AttachedCollisionObject.from_attached_collision_mesh(acm)
-            start_state.attached_collision_objects.append(aco)
+
+        if options.get('attached_collision_meshes'):
+            for acm in options['attached_collision_meshes']:
+                aco = AttachedCollisionObject.from_attached_collision_mesh(acm)
+                start_state.attached_collision_objects.append(aco)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py:129]: that's exactly what it was two or days ago, but i decided to be clever and reduce indentation.

**Review by beverlylytle — CHANGES_REQUESTED** [review]
I think you ought to take a look at robot.py too. `inverse_kinematics`, etc...

File: src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py:129]: ;)

**Comment by gonzalocasas** [comment]
> I think you ought to take a look at robot.py too. `inverse_kinematics`, etc...

Fixed a bunch more of these here: https://github.com/compas-dev/compas_fab/pull/269/commits/c61ff578328a8b0cf7b233059e3686302d016ff8.
I hope that's all of them, I did a regex search across the code base for `\.get\(.*,`.

File: src/compas_fab/backends/pybullet/backend_features/pybullet_inverse_kinematics.py [hunk 2]
```diff
@@ -77,7 +77,9 @@ def inverse_kinematics(self, robot, frame_WCF, start_configuration=None, group=N
             start_configuration = self.client.set_robot_configuration(robot, start_configuration, group)
 
         called_from_test = 'pytest' in sys.modules
-        if options.get('enforce_joint_limits', True) and not called_from_test:
+        enforce_joint_limits = options.get('enforce_joint_limits') or True
```
Comment (beverlylytle) [inline src/compas_fab/backends/pybullet/backend_features/pybullet_inverse_kinematics.py:80]: This one's no good at all.  It forces `enforce_joint_limits` to never be `False`. Maybe something like
```suggestion
        enforce_joint_limits = True if options.get('enforce_joint_limits') is None or options['enforce_joint_limits']
```

File: src/compas_fab/backends/ros/backend_features/move_it_forward_kinematics.py [hunk 3]
```diff
@@ -67,10 +67,10 @@ def forward_kinematics(self, robot, configuration, group=None, options=None):
         kwargs['errback_name'] = 'errback'
 
         # Use base_link or fallback to model's root link
-        options['base_link'] = options.get('base_link', robot.model.root.name)
+        options['base_link'] = options.get('base_link') or robot.model.root.name
 
         # Use selected link or default to group's end effector
-        options['link'] = options.get('link', options.get('ee_link')) or robot.get_end_effector_link_name(group)
+        options['link'] = options.get('link') or options.get('ee_link') or robot.get_end_effector_link_name(group)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_forward_kinematics.py:73]: This is much better than it was.

Comment (beverlylytle) [reply src/compas_fab/backends/ros/backend_features/move_it_forward_kinematics.py:73]: Also, I suppose (hope) it isn't valid to have an empty string as a link name?

File: src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py [hunk 4]
```diff
@@ -112,8 +112,8 @@ def inverse_kinematics_async(self, callback, errback,
                                        robot_state=start_state,
                                        constraints=constraints,
                                        pose_stamped=pose_stamped,
-                                       avoid_collisions=options.get('avoid_collisions', True),
-                                       attempts=options.get('attempts', 8))
+                                       avoid_collisions=options.get('avoid_collisions') or True,
+                                       attempts=options.get('attempts') or 8)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:116]: I suppose it's silly to have zero attempts, but...

File: src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py [hunk 5]
```diff
@@ -112,8 +112,8 @@ def inverse_kinematics_async(self, callback, errback,
                                        robot_state=start_state,
                                        constraints=constraints,
                                        pose_stamped=pose_stamped,
-                                       avoid_collisions=options.get('avoid_collisions', True),
-                                       attempts=options.get('attempts', 8))
+                                       avoid_collisions=options.get('avoid_collisions') or True,
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py:115]: Same as with `enforce_joint_limits` as above.

File: src/compas_fab/backends/ros/backend_features/move_it_plan_cartesian_motion.py [hunk 6]
```diff
@@ -128,9 +128,9 @@ def plan_cartesian_motion_async(self, callback, errback,
                        group_name=group,
                        link_name=options['link'],
                        waypoints=waypoints,
-                       max_step=float(options.get('max_step', 0.01)),
-                       jump_threshold=float(options.get('jump_threshold', 1.57)),
-                       avoid_collisions=bool(options.get('avoid_collisions', True)),
+                       max_step=float(options.get('max_step') or 0.01),
+                       jump_threshold=float(options.get('jump_threshold') or 1.57),
+                       avoid_collisions=bool(options.get('avoid_collisions') or True),
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_plan_cartesian_motion.py:133]: Again, trouble...

File: src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py [hunk 7]
```diff
@@ -140,12 +140,12 @@ def plan_motion_async(self, callback, errback,
...
+                       planner_id=options.get('planner_id') or 'RRTConnectkConfigDefault',
                        group_name=group,
-                       num_planning_attempts=options.get('num_planning_attempts', 1),
-                       allowed_planning_time=options.get('allowed_planning_time', 2.),
-                       max_velocity_scaling_factor=options.get('max_velocity_scaling_factor', 1.),
-                       max_acceleration_scaling_factor=options.get('max_acceleration_scaling_factor', 1.))
+                       num_planning_attempts=options.get('num_planning_attempts') or 1,
+                       allowed_planning_time=options.get('allowed_planning_time') or 2.,
+                       max_velocity_scaling_factor=options.get('max_velocity_scaling_factor') or 1.,
+                       max_acceleration_scaling_factor=options.get('max_acceleration_scaling_factor') or 1.)
```
Comment (beverlylytle) [inline src/compas_fab/backends/ros/backend_features/move_it_plan_motion.py:148]: Does zero make sense for these? I suppose not, but just checking.

File: src/compas_fab/backends/vrep/backend_features/vrep_inverse_kinematics.py [hunk 8]
```diff
@@ -50,7 +50,7 @@ def inverse_kinematics(self, robot, frame_WCF, start_configuration=None, group=N
         gantry_joint_limits = options.get('gantry_joint_limits')
         arm_joint_limits = options.get('arm_joint_limits')
         max_trials = options.get('max_trials')
-        max_results = options.get('max_results', 1)
+        max_results = options.get('max_results') or 1
```
Comment (beverlylytle) [inline src/compas_fab/backends/vrep/backend_features/vrep_inverse_kinematics.py:53]: Zero?

File: src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py [hunk 9]
```diff
@@ -67,13 +67,13 @@ def plan_motion(self, robot, goal_constraints, start_configuration=None, group=N
         num_joints = len(robot.get_configurable_joints())
         metric_values = options.get('metric_values')
         collision_meshes = options.get('collision_meshes')
-        planner_id = options.get('planner_id', 'rrtconnect')
-        trials = options.get('trials', 1)
-        resolution = options.get('resolution', 0.02)
+        planner_id = options.get('planner_id') or 'rrtconnect'
+        trials = options.get('trials') or 1
+        resolution = options.get('resolution') or 0.02
```
Comment (beverlylytle) [inline src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py:72]: Zero?

File: src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py [hunk 10]
```diff
@@ -67,13 +67,13 @@ def plan_motion(self, robot, goal_constraints, start_configuration=None, group=N
...
         gantry_joint_limits = options.get('gantry_joint_limits')
         arm_joint_limits = options.get('arm_joint_limits')
-        shallow_state_search = options.get('shallow_state_search', True)
-        optimize_path_length = options.get('optimize_path_length', False)
+        shallow_state_search = options.get('shallow_state_search') or True
```
Comment (beverlylytle) [inline src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py:75]: trouble

File: src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py [hunk 11]
```diff
@@ -124,13 +124,13 @@ def plan_motion_to_config(self, robot, goal_configs, start_configuration=None, g
         num_joints = len(robot.get_configurable_joints())
         metric_values = options.get('metric_values')
         collision_meshes = options.get('collision_meshes')
-        planner_id = options.get('planner_id', 'rrtconnect')
-        trials = options.get('trials', 1)
-        resolution = options.get('resolution', 0.02)
+        planner_id = options.get('planner_id') or 'rrtconnect'
+        trials = options.get('trials') or 1
+        resolution = options.get('resolution') or 0.02
```
Comment (beverlylytle) [inline src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py:129]: zero?

File: src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py [hunk 12]
```diff
@@ -124,13 +124,13 @@ def plan_motion_to_config(self, robot, goal_configs, start_configuration=None, g
...
         gantry_joint_limits = options.get('gantry_joint_limits')
         arm_joint_limits = options.get('arm_joint_limits')
-        shallow_state_search = options.get('shallow_state_search', True)
-        optimize_path_length = options.get('optimize_path_length', False)
+        shallow_state_search = options.get('shallow_state_search') or True
```
Comment (beverlylytle) [inline src/compas_fab/backends/vrep/backend_features/vrep_plan_motion.py:132]: trouble

**Review by beverlylytle — COMMENTED** [review]
robot.py has the line `options.get('attached_collision_meshes', [])` in `inverse_kinematics`, `plan_motion` and `plan_cartesian_motion`.

**Comment by gonzalocasas** [comment]
> robot.py has the line `options.get('attached_collision_meshes', [])` in `inverse_kinematics`, `plan_motion` and `plan_cartesian_motion`.

that is fixed already, on the first commit. i think ;)

**Comment by beverlylytle** [comment]
> > robot.py has the line `options.get('attached_collision_meshes', [])` in `inverse_kinematics`, `plan_motion` and `plan_cartesian_motion`.
> 
> that is fixed already, on the first commit. i think ;)

oh, some how that wasn't the first diff i was looking at.

**Comment by gonzalocasas** [comment]
Should I revert the second commit with the various other fixes? It smells like more problems than solutions...

**Comment by beverlylytle** [comment]
> Should I revert the second commit with the various other fixes? It smells like more problems than solutions...

yeah, i think for a hot fix, this is the better option.  when there's more time to investigate the other potential sources of bugs, they can be addressed then.

**Review by beverlylytle — APPROVED** [review]

**Comment by gonzalocasas** [comment]
done! let's see if tests are happy. The `robot.py` changes should be in. For a proper fix, as you (gladly spotted on time) this needs to be carefully done in all cases were a valid value of the option is a falsy value.

**Comment by beverlylytle** [comment]
tests are happy!

**Pull request merged**